PGM/src/main/java/tc/oc/pgm/map/MapPersistentContext.java

```diff
@@ -0,0 +1,48 @@
+package tc.oc.pgm.map;
+
+import com.google.common.collect.Range;
+import org.jdom2.Document;
+import tc.oc.api.docs.SemanticVersion;
+import tc.oc.pgm.modules.InfoModule;
+
+/**
+ * This is a replacement for classes that need information from MapModuleContext to use. MapModuleContext will not be
+ * guaranteed to be available if a match isn't being played for the map, but MapPersistentContext will always contain
+ * basic information required to reference the map (like map name, authors, or player amount).
+ */
+public class MapPersistentContext {
+
+    private SemanticVersion proto;
+    private Range<Integer> playerLimits;
+    private InfoModule infoModule;
+    private MapDocument apiDocument;
+
+    public MapPersistentContext(MapModuleContext context) {
+        this.proto = context.getProto();
+        this.playerLimits = context.playerLimits();
+        this.infoModule = context.needModule(InfoModule.class);
+        this.apiDocument = context.apiDocument();
```

Comment:
Is creating the API doc here before it's requested an issue?

Reply:
Probably. Looking at `MapModuleContext` it has an injected `Provider<MapDocument>`. I would expose that provider as a method so you can retrieve and store it to bypass needing the whole context object.

Reply:
Also, the MapDocument needs to be built in the MapScope, MapModuleContext has this easy as it's the store for the scope, so it can just run `asCurrentScope(apiDocumentProvider::get);` but the MapPersistentContext doesn't have direct access to the injectionStore (the MapContext) and it may not even have access anymore if the context has been unloaded.